File: python/leetcode102.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if not root:
            return []
        layer = [root]
        res = [[root.val]]
        while len(layer) > 0:
            tmp = []
            for x in layer:
                if x.left:
                    tmp.append(x.left)
                if x.right:
                    tmp.append(x.right)
            layer = tmp
            if len(layer) > 0:
                res.append([x.val for x in layer])
        return res

File: leetcodec++/leetcode102.cpp
//
// Created by 黄敏辉 on 2019-07-31.
//

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    vector<vector<int >> levelOrder(TreeNode* root) {
        vector<vector<int >> res;
        vector<TreeNode*> nodes;
        if (!root)
            return res;
        nodes.push_back(root);
        while(!nodes.empty()){
            vector<TreeNode*> tmp;
            vector<int > lel;
            for (int i = 0; i < nodes.size(); i++){
                lel.push_back(nodes[i]->val);
                if (nodes[i]->left)
                    tmp.push_back(nodes[i]->left);
                if (nodes[i]->right)
                    tmp.push_back(nodes[i]->right);
            }
            res.push_back(lel);
            nodes = tmp;
        }
        return res;
    }
};